### PR TITLE
Neoskye refactor

### PR DESCRIPTION
**Pull request by cyon1c**

### CODE REVIEW

File: pdn-code/include/comms.hpp [hunk 1]
```diff
@@ -0,0 +1,27 @@
+#pragma once
```
[the remark this answers is not in the document]
Comment (NeoSkye) [reply pdn-code/include/comms.hpp:1]: This prevents the header from being included multiple times (often called a header guard). It's the modern variant of:

```
#ifndef _MYHEADER_H_
#define _MYHEADER_H_

<header code>

#endif //_MYHEADER_H_

```

File: pdn-code/src/main.cpp [hunk 2]
```diff
@@ -128,27 +129,9 @@ void setTransmitLight(boolean on);
...
-void writeDebugByte(byte command);
-
-void monitorTX();
-void monitorRX();
-
-void writeTxTransaction(byte command);
-void writeRxTransaction(byte command);
-
+//FIXME: This requires stream validation so can't easily
+//be moved to separate file yet, but should eventually be.
```
[the remark this answers is not in the document]
Comment (NeoSkye) [reply pdn-code/src/main.cpp:133]: `flushComms()` potentially calls into `isValidMessageSerial1()` or `isValidMessageSerial2()` which rely on getting player information data. We need a safe way of sharing that data between the files.

You'll soon see a commit on this branch that creates a PlayerInfo class. Once I start encapsulating the comms functionality, that will be able to share a pointer to the main PlayerInfo class, allowing this to be moved to the comms encapsulation.

File: pdn-code/include/comms.hpp [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+String readDebugString(char terminator);
+byte peekDebugComms();
+void writeDebugByte(byte command);
+
+// FIXME: These appear to be unused so should be safe to remove
+// but it would be best for whomever first added them to be consulted
+// if possible.
+
+//void monitorTX();
+//void monitorRX();
```
[the remark this answers is not in the document]
Comment (NeoSkye) [reply pdn-code/include/comms.hpp]: Cool, removed in b80176259e5b5fd7241a81e2d67a4a16baff41f8 on this branch.

**Comment by NeoSkye** [comment]
Pushed some additional refactor, mostly encapsulating globals into classes in separate files.

Bare minimum of testing done so far. Definitely needs more testing before we consider a merge (plus I have more refactor work I'd like to do before we merge).

File: pdn-code/src/main.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (cyon1c) [reply pdn-code/src/main.cpp:128]: Cool! Looking forward to it.

**Comment by NeoSkye** [comment]
Finished refactor for now. This is still a bit messy, but a lot of it will fix itself with the state machine, so I think this is a good stopping/merge point.

**Pull request merged**